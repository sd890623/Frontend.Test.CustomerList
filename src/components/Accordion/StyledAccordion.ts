import styled from "styled-components";

export const StyledAccordion = styled.div`
  width:100%;
  display: block;
  text-align: center;
`;

export const StyledAccordionButton = styled.button<{isOpen: boolean}>`
  background: ${prop=>prop.isOpen ? 'white' : '#ccffff'};
  cursor: pointer;
  font-size: 1rem;
  padding: 1rem;
  width:100%;
`;

export const StyledAccordionContent = styled.div<{isOpen: boolean}>`
  padding-top:${prop=>prop.isOpen ? '1rem' : '0'};
  overflow: hidden;
  transition: all 0.3s ease-in-out;
  max-height: ${prop=>prop.isOpen ? '350px' : '0'};
  text-align: left;
`;
