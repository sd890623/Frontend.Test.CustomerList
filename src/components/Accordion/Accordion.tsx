import {FC, ReactNode, useState} from "react";
import { StyledAccordion, StyledAccordionButton, StyledAccordionContent } from "./StyledAccordion";

interface Props {
    title: string;
    children: ReactNode
}

const Accordion: FC<Props> = (props:Props)=>{
    const [isOpen,setOpen]=useState(false)
    return (
        <StyledAccordion>
            <StyledAccordionButton isOpen={isOpen} onClick={()=>setOpen((state=>!state))} data-testid="button">
                {props.title}
            </StyledAccordionButton>
            <StyledAccordionContent isOpen={isOpen} data-testid="content" className={isOpen ? 'open' : 'closed'}>
                {props.children}
            </StyledAccordionContent>
        </StyledAccordion>
    )
}
export default Accordion