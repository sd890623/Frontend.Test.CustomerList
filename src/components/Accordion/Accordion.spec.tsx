import * as React from "react"
import {fireEvent} from '@testing-library/react';
import { render } from "../../utils/testUtils";
import Accordion from "./Accordion";

describe("<Accordion />", () => {
    it("should render a <Accordion />", () => {
        const wrapper = render(<Accordion title="accordionTitle"><p>Content</p></Accordion>);
        expect(wrapper.container).toMatchSnapshot();
    })
    it("should toggle the content in or out by clicking button", () => {
        const { getByTestId }= render(<Accordion title='accordionTitle'><p>Content</p></Accordion>);
        expect(getByTestId('content')).toBeInTheDocument();
        expect(getByTestId('content')).toHaveClass('closed')
        fireEvent.click(getByTestId('button'))
        expect(getByTestId('content')).toBeInTheDocument();
        expect(getByTestId('content')).toHaveClass('open')
    })
    // @todo better testing for hidden by height-visibility
});
