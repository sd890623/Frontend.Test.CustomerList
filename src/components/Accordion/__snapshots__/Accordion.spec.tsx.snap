// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render a <Accordion /> 1`] = `
<div>
  <div
    class="sc-bdnxRM bHJSvJ"
  >
    <button
      class="sc-gtsrHT NeZzs"
      data-testid="button"
    >
      accordionTitle
    </button>
    <div
      class="sc-dlnjwi idkNxN closed"
      data-testid="content"
    >
      <p>
        Content
      </p>
    </div>
  </div>
</div>
`;
